### ISSUE TITLE
[Improvement]: xml vehicles

### ISSUE THREAD

**Issue by n8tr0** [YimMenu/YimMenu#2391]
### Which feature

Added support for more modifications in "xml_vehicles".
Example:
-Livery
-WheelType
-LrInterior
-DoorsBroken
-NumberPlateText
-NumberPlateIndex
-Neons
etc..

Most of these options already work well in "xml_maps" I don't think adding support for these options to "xml_vehicles" would be too complicated.

### Reason

At present it is not possible to load a vehicle with a livery, rims etc..

### Additional context

I imagine it's not something very important for many, but having to manually modify the vehicle after loading is rather annoying.
If anyone finds the time and inclination to set this up, I'd like to thank them in advance.

**Comment by alyxme** [comment]
Related to #1852 scenarios are also ignored

**Comment by gir489returns** [comment]
Do those scenarios exist in Menyoo? The XML Vehicles code was only meant to really support legacy Menyoo vehicles, if you are looking to properly persist those vehicles to disk, use the JSON persist engine, as it is much more robust.

**Comment by alyxme** [comment]
Yes they do in Menyoo! [Example.xml](https://github.com/YimMenu/YimMenu/files/12168556/xml_example.zip) with scenario `<ScenarioName>WORLD_HUMAN_SECURITY_SHINE_TORCH</ScenarioName>`. This xml file is from GTA5-mods section for Menyoo. I described in #1852 my findings about that. 
Perhaps this may also be relevant: if there are several peds with the same animations in the xml file, then some of them are not played. I previously experimented with fiberpool and it seemed to help tp play animations.
Unfortunately, I was unable to get the code to fix it.

**Comment by n8tr0** [comment]
> Do those scenarios exist in Menyoo? The XML Vehicles code was only meant to really support legacy Menyoo vehicles, if you are looking to properly persist those vehicles to disk, use the JSON persist engine, as it is much more robust.

JSON persist engine works well, but unfortunately it lacks a few functions, such as "Bone Attachment" "Doors Broken" and probably other things I can't think of.
